### ISSUE TITLE
bug: cannot start localstack with Docker user namespaces enabled 

### ISSUE THREAD

**Issue by smcoll** [localstack/localstack#6771]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

i am unable to start localstack via the CLI on a machine where Docker is configured with user namespaces.

### Expected Behavior

_No response_

### How are you starting LocalStack?

With the `localstack` script

### Steps To Reproduce

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

```
$ localstack start
     __                     _______ __             __
    / /   ____  _________ _/ / ___// /_____ ______/ /__
   / /   / __ \/ ___/ __ `/ /\__ \/ __/ __ `/ ___/ //_/
  / /___/ /_/ / /__/ /_/ / /___/ / /_/ /_/ / /__/ ,<
/_____/\____/\___/\__,_/_//____/\__/\__,_/\___/_/|_|
 💻 LocalStack CLI 1.0.4
[10:08:07] starting LocalStack in Docker mode 🐳                                                                                                                                                                                                                localstack.py:138
───────────────────────────────────────────────────────────────────────────────────────────────────────────────── LocalStack Runtime Log (press CTRL-C to quit) ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────
2022-08-28T10:08:08.172 ERROR --- [    Thread-2] localstack.utils.bootstrap : Error while starting LocalStack container: Docker process returned with errorcode 125
docker: Error response from daemon: privileged mode is incompatible with user namespaces.  You must run the container in the host namespace when running privileged mode.
See 'docker run --help'.
```

In debug mode, i found that the CLI is running this docker command:
```
docker run --rm --name localstack_main -v /home/me/.cache/localstack/volume:/var/lib/localstack -v /var/run/docker.sock:/var/run/docker.sock -p 127.0.0.1:4566:4566 -p 127.0.0.1:4571:4571 -p 127.0.0.1:12121:12121 -p 127.0.0.1:4510-4559:4510-4559 -e TEST_AWS_ACCOUNT_ID=000000000000 -e DEFAULT_REGION=us-east-1 -e DEBUG=1 -e DOCKER_HOST=unix:///var/run/docker.sock -e HOST_TMP_FOLDER=/home/me/.local/lib/python3.9/site-packages/.filesystem/var/lib/localstack/tmp -e SET_TERM_HANDLER=1 --privileged localstack/localstack
```
i am able to run that command only if i remove the `--privileged` flag. But i don't know why localstack might need that enabled in the first place, and don't see a way to configure localstack CLI to launch without that flag set.



### Environment

```markdown
- OS: Ubuntu 21.10 ... will be updated shortly
- LocalStack: 1.0.4
```


### Anything else?

For what it's worth, i don't see how to get an account at https://localstack-community.slack.com/

**Comment by steffyP** [comment]
Hi @smcoll, 

> For what it's worth, i don't see how to get an account at https://localstack-community.slack.com/

You can find details [here on our website](https://localstack.cloud/community/get-involved/).

Do you still experience the issue you reported?

**Comment by smcoll** [comment]
i do still have this issue.

Also, at https://localstack-community.slack.com/signup#/domain-signup my google SSO is not accepted ("The email address must match one of the domains listed below. Please try another email."). i do not have a localstack.cloud account either.

**Comment by steffyP** [comment]
@smcoll, regarding the priviliged mode issue: could you please update the LocalStack cli? 🙏 

We included a fix in LocalStack version `1.2.0`, but would recommend to use the latest `localstack` cli `2.2.0`, you can see the [release history on pypi](https://pypi.org/project/localstack/#history).

Additionally, you should update the LocalStack images. You could do so using the cli, e.g. `localstack update docker-images` or pulling the images manually, e.g. `docker pull localstack/localstack:latest`.  

Let us know if this fixes your issue 🙏

**Comment by smcoll** [comment]
i have the same issue:

```
$ localstack --version
2.2.0
$ localstack update docker-images
───────────────────────────────────── Updating docker images ──────────────────────────────────────
✔ Image public.ecr.aws/lambda/python:3.9 updated.
✔ Image localstack/localstack:0.12.18 up-to-date.
✔ Image lambci/lambda:python3.8 up-to-date.
✔ Image lambci/lambda:python3.7 up-to-date.
✔ Image lambci/lambda:python2.7 up-to-date.
───────────────────────────────────────────────────────────────────────────────────────────────────
Images updated: 1, Images failed: 0, total images processed: 5.

$ localstack start

     __                     _______ __             __
    / /   ____  _________ _/ / ___// /_____ ______/ /__
   / /   / __ \/ ___/ __ `/ /\__ \/ __/ __ `/ ___/ //_/
  / /___/ /_/ / /__/ /_/ / /___/ / /_/ /_/ / /__/ ,<
 /_____/\____/\___/\__,_/_//____/\__/\__,_/\___/_/|_|

 💻 LocalStack CLI 2.2.0

[14:21:26] starting LocalStack in Docker mode 🐳                                  localstack.py:409
────────────────────────── LocalStack Runtime Log (press CTRL-C to quit) ──────────────────────────
2023-09-09T14:21:26.738 ERROR --- [-functhread2] localstack.utils.bootstrap : Error while starting LocalStack container: Docker process returned with errorcode 125
docker: Error response from daemon: privileged mode is incompatible with user namespaces.  You must run the container in the host namespace when running privileged mode.
See 'docker run --help'.

Error: Docker process returned with errorcode 125
```

**Comment by steffyP** [comment]
Hi @smcoll, 

can you please run `docker pull localstack/localstack:latest`? 
The `update docker-images` only updates images that have been used at least once before. From the logs it seems like you have used version `0.12.18` only, and we do have any fixes for that version. 
Please update to a newer version 🙏  You can also check our release notes and tips for upgrading in [LocalStack Discuss](https://discuss.localstack.cloud/tags/c/announcement/5/release).

**Comment by smcoll** [comment]
@steffyP same output:

```
$ docker pull localstack/localstack:latest
latest: Pulling from localstack/localstack
7dbc1adf280e: Extracting [========>                                          ]  5.243MB/31.42MB7dbc1adf280e: Pull complete 
4171d440ff02: Pull complete 
1bde28236607: Pull complete 
28f4c351f921: Pull complete 
78fa60ffcbfa: Pull complete 
cb187d83b7fc: Pull complete 
388b73f3e9a7: Pull complete 
1331030ce478: Pull complete 
288628244005: Pull complete 
ca8b2de5f22f: Pull complete 
d42f9ea5bb61: Pull complete 
a72eefbe3022: Pull complete 
66eccd8f09fa: Pull complete 
4f4fb700ef54: Pull complete 
54e3f24018a4: Pull complete 
fb53f05e8d84: Pull complete 
31217248fb1e: Pull complete 
ed6151e5a401: Pull complete 
135f79195a37: Pull complete 
fbf1c5c7d295: Pull complete 
28b66ff751fc: Pull complete 
bdd61baa9452: Pull complete 
503b260a5cf8: Pull complete 
537059bb448f: Pull complete 
83f48826023e: Pull complete 
15924e20ab54: Pull complete 
029f8c26e23b: Pull complete 
18f0f912c84c: Pull complete 
7799d3626a8a: Pull complete 
Digest: sha256:a6f60e0950b99df4098a7979a9070b4b0a27faac0bad99b55e5404a9275fe668
Status: Downloaded newer image for localstack/localstack:latest
docker.io/localstack/localstack:latest
$ localstack --version
2.2.0
$ localstack update docker-images
────────────────────────────────────────────────────────── Updating docker images ──────────────────────────────────────────────────────────
✔ Image localstack/localstack:latest up-to-date.
✔ Image public.ecr.aws/lambda/python:3.9 updated.
✔ Image localstack/localstack:0.12.18 up-to-date.
✔ Image lambci/lambda:python3.8 up-to-date.
✔ Image lambci/lambda:python3.7 up-to-date.
✔ Image lambci/lambda:python2.7 up-to-date.
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Images updated: 1, Images failed: 0, total images processed: 6.
$ localstack start

     __                     _______ __             __
    / /   ____  _________ _/ / ___// /_____ ______/ /__
   / /   / __ \/ ___/ __ `/ /\__ \/ __/ __ `/ ___/ //_/
  / /___/ /_/ / /__/ /_/ / /___/ / /_/ /_/ / /__/ ,<
 /_____/\____/\___/\__,_/_//____/\__/\__,_/\___/_/|_|

 💻 LocalStack CLI 2.2.0

[16:20:30] starting LocalStack in Docker mode 🐳                                                                           localstack.py:409
────────────────────────────────────────────── LocalStack Runtime Log (press CTRL-C to quit) ───────────────────────────────────────────────
2023-10-01T16:20:30.960 ERROR --- [-functhread2] localstack.utils.bootstrap : Error while starting LocalStack container: Docker process returned with errorcode 125
docker: Error response from daemon: privileged mode is incompatible with user namespaces.  You must run the container in the host namespace when running privileged mode.
See 'docker run --help'.

Error: Docker process returned with errorcode 125
```

**Comment by simonrw** [comment]
Could you update your `localstack` cli to the latest version? The fix should be included in version 2.3.

* if you downloaded the cli, could you re-download the latest version?
* if you used `pip` could you reinstall with `pip install -U localstack`?

**Comment by simonrw** [comment]
Unfortunately using LocalStack in a user namespace means limited functionality since LocalStack cannot make any requests to the docker daemon. We rely on the docker daemon for quite a few pieces of functionality. In particular, lambda functions, testing port binding (that is the cause of the warning), documentdb etc. If you do not need these features then you should still be able to use LocalStack.

There may be some workarounds available to you. From an initial search: https://stackoverflow.com/questions/44762963/share-docker-socket-using-user-namespaces
